backend/src/crawler/udn_crawler.py

```diff
@@ -0,0 +1,143 @@
+    _extract_news(soup, url: str) -> News: Extracts news details from the BeautifulSoup object.
+    save(self, news: News, db: Session): Saves a news article to the database.
+    _commit_changes(db: Session): Commits the changes to the database with error handling.
+"""
+
+from requests import Response, get
+from bs4 import BeautifulSoup
+from sqlalchemy.orm import Session
+from urllib.parse import quote
+
+from .crawler_base import NewsCrawlerBase, Headline, News, NewsWithSummary
+from ..news.models import NewsArticle
+
+class UDNCrawler(NewsCrawlerBase):
+    CHANNEL_ID = 2
+
+    def __init__(self, timeout: int = 5) -> None:
+        self.news_website_url = "https://udn.com/api/more"
+        self.timeout = timeout
+
+    def startup(self, search_term: str) -> list[Headline]:
+        """
+        Initializes the application by fetching news headlines for a given search term across multiple pages.
+        This method is typically called at the beginning of the program when there is no data available,
+        hence it fetches headlines from the first 10 pages.
+
+        :param search_term: The term to search for in news headlines.
+        :return: A list of Headline namedtuples containing the title and URL of news articles.
+        :rtype: list[Headline]
+        """
+        return self.get_headline(search_term, page=(1, 10))
+
+    def get_headline(
+        self, search_term: str, page: int | tuple[int, int]
+    ) -> list[Headline]:
+
+        # Calculate the range of pages to fetch news from.
+        # If 'page' is a tuple, unpack it and create a range representing those pages (inclusive).
+        # If 'page' is an int, create a list containing only that single page number.
+        # page_range = range(*page) if isinstance(page, tuple) else [page]
+
+        # Calculate the range of pages to fetch news from.
+        # If 'page' is a tuple, unpack it and create a range representing those pages (inclusive).
+        # If 'page' is an int, create a list containing only that single page number.
+        
+        page_range = range(*page) if isinstance(page, tuple) else [page]
+        news_data = []
+        for page in page_range:
+            news_data.extend(self._fetch_news(page, search_term))
+
+        return news_data
+
+    def _fetch_news(self, page: int, search_term: str) -> list[Headline]:
+        response = self._perform_request(self.news_website_url,
+                                         self._create_search_params(page, search_term, "searchword"))
+        return self._parse_headlines(response)
+
+    def _create_search_params(self, page: int, search_term: str) -> dict:
+        return {
+            "page": page,
+            "id": f"search:{quote(search_term)}",
+            "channelId": self.CHANNEL_ID,
+            "type": "searchword"
+        }
+
+    def _perform_request(self, url: str | None = None, params: dict | None = None) -> Response:
+        return get(url, params=params)
```

Comment:
The initializer of UDNCrawler accepts the `timeout` parameter for HTTP requests, but it seems like the
`_perform_request` never performs HTTP requests with the `timeout` property.

```suggestion
        return get(url, params=params, timeout=self.timeout)
```